pkg/vault/test/foundry/ProtocolFeeController.t.sol

```diff
@@ -1320,7 +1336,44 @@ contract ProtocolFeeControllerTest is BaseVaultTest {
         vm.expectEmit();
         emit IProtocolFeeController.InitialPoolAggregateYieldFeePercentage(pool, MAX_PROTOCOL_YIELD_FEE_PCT, false);
 
-        PoolFactoryMock(poolFactory).registerTestPool(pool, tokenConfig);
+        vm.expectEmit();
+        emit IProtocolFeeController.PoolRegisteredWithFeeController(pool, lp, false);
+
+        PoolRoleAccounts memory roleAccounts;
+        roleAccounts.poolCreator = lp;
+
+        PoolFactoryMock(poolFactory).registerGeneralTestPool(
+            pool,
+            tokenConfig,
+            0,
+            365 days,
+            false, // not exempt from protocol fees
+            roleAccounts,
+            address(0)
+        );
+
+        assertTrue(feeController.isPoolRegistered(pool), "Pool not registered");
+    }
+
+    function testPoolRegistrationInvalid() public view {
+        assertFalse(feeController.isPoolRegistered(ZERO_ADDRESS), "Invalid pool registered");
+    }
+
+    function testRegisterExistingPoolInMigration() public {
+        _registerPoolWithMaxProtocolFees();
+
+        ProtocolFeeController newFeeController = new ProtocolFeeController(vault);
+
+        ProtocolFeeController(address(newFeeController)).migratePool(pool);
+
+        vm.expectRevert(abi.encodeWithSelector(ProtocolFeeController.PoolAlreadyRegistered.selector, pool));
+        ProtocolFeeController(address(newFeeController)).migratePool(pool);
```

Comment:
I'd also check that the values set for the original pool are actually there in the new fee controller:
- swap protocol fee / is override
- yield protocol fee / is override
- swap creator fee
- yield creator fee